services/gateway/gateway-service.mjs

```diff
@@ -310,24 +346,22 @@ class GatewayService {
     return true;
   }
 
-  _getDistanceSquare(a,b){
-    return Math.pow(Math.abs(a.x - b.x),2) + Math.pow(Math.abs(a.y - b.y),2);
+  _getManhattanDistance(a,b){
+    return Math.abs(a.x - b.x) + Math.abs(a.y - b.y);
   }
 
-  _borderCheck(msg,mapSize){
-    if(msg.x > mapSize.width || msg.x < 0){
+  _borderCheck(coord,mapSize){
+    if(coord.x > mapSize.width || coord.x < 0){
       return false;
     }
-    if(msg.y > mapSize.height || msg.y < 0){
+    if(coord.y > mapSize.height || coord.y < 0){
       return false;
     }
     return true;
   }
 
-  _nearByGridCheck(msg,lastRecord){
-    let lastPosition = {x:lastRecord.x , y:lastRecord.y};
-    let targetPosition = {x:msg.x , y:msg.y};
-    let distanceSquire = this._getDistanceSquare(lastPosition,targetPosition);
+  _nearByGridCheck(coord,lastCoord){
```

Comment:
_**Nearby**_ is a word, so maybe it should be `_nearbyGridCheck`

Reply:
I don't see the need for this function. Isn't the functionality already been included in the speed check?

Reply:
This is needed because speed check checks only the time, this checks the distance.

Reply:
The definition of speed is distance divided by time. The speed check should check whether a player **_moves how far_** within **_how much time_**.